lib/providers/config_provider.dart

```diff
@@ -1,142 +1,370 @@
+  ConnectionPreferences get currentConnection => _settingsPreferences.currentConnection;
+
   int get snackBarsToRemove => _snackBarsToRemove;
 
+  bool get webSocketConnectionErrorPresent => _webSocketConnectionErrorPresent;
+
+  String get notificationPayload => _notificationPayload;
+
+  Board get currentBoard => _currentBoard;
+
+  int get notificationFrequency => _settingsPreferences.notificationFrequencyInMinutes;
+
+  DateTime get lastNotificationTimestamp => _settingsPreferences.lastNotificationTimestamp;
+
+  get showNotifications => _settingsPreferences.showNotifications;
+
+  get showHints => _settingsPreferences.showHints;
+
+  Map<String, bool> get hints {
+    return _settingsPreferences.hints;
+  }
+
   List<DashboardWidget> get favouriteWidgets {
-    return _config.widgets.widgetsById.entries
-        .map((entry) => entry.value)
-        .where((widget) => _urlPreferences.favouriteWidgetIds.contains(widget.id))
-        .toList();
+    return getSortedWidgetsList(currentConnection.favouriteWidgets);
   }
 
   List<DashboardWidget> get quarantineWidgets {
-    return _config.widgets.widgetsById.entries
-        .map((entry) => entry.value)
-        .where((widget) => _urlPreferences.quarantineWidgetIds.contains(widget.id))
-        .toList();
+    return getSortedWidgetsList(currentConnection.quarantineWidgets);
+  }
+
+  bool isWidgetInQuarantine(DashboardWidget widget) {
+    return quarantineWidgets.indexWhere((element) => element.id == widget.id) > -1;
   }
 
   List<Board> get boards {
     return [..._boards];
   }
 
-  Future<void> fetchConfig() async {
-    final response = await http.get(_currentUrl);
-    _config = Config.fromJson(json.decode(response.body) as Map<String, dynamic>);
-    _boards = _config.boards.boardsById.entries.map((entry) => entry.value).toList();
-    if (await SharedPref.containsKey(_currentUrl)) {
-      _urlPreferences = UrlPreferences.fromJson(jsonDecode(await SharedPref.read(_currentUrl)));
-    }
-    notifyListeners();
+  List<DashboardWidget> getAllWidgets() {
+    return _config.widgets.widgetsById.entries.map((entry) => entry.value).toList();
+  }
+
+  List<DashboardWidget> getAllWidgetsDeepCopy() {
+    return new List<DashboardWidget>.from(getAllWidgets().map((widget) => DashboardWidget.deepCopy(widget)).toList());
+  }
+
+  String getWidgetName(DashboardWidget widget) {
+    return (widget.title != null && widget.title.isNotEmpty) ? widget.title : widget.type;
   }
 
   List<DashboardWidget> getBoardWidgets(Board board) {
-    return _config.widgets.widgetsById.entries
-        .map((entry) => entry.value)
-        .where((widget) => board.widgets.contains(widget.id))
-        .toList();
+    List<DashboardWidget> boardWidgets = [
+      ..._config.widgets.widgetsById.entries
+          .map((entry) => entry.value)
+          .where((widget) =>
+              board.widgets.contains(widget.id) &&
+              quarantineWidgets.indexWhere((element) => element.id == widget.id) == -1 &&
```

Comment:
🥇 [NOW] there is a method for this in line #120

Reply:
done